include/core/parameters.h

```diff
@@ -856,6 +881,20 @@ namespace Parameters
       temperature
     } variable;
 
+    // WIP Map containing the refinement variables
+    std::map<Variable, MultipleAdaptationParameters> variables;
+    // declaration for parsing variables
+    Variable                     vars;
+    MultipleAdaptationParameters var_adaptation_param;
```

Comment:
I know this is something you want to remove eventually, but you could actually remove it right now by already using the key of the map (assuming there is only one). This way generallisation will be quasi-done.